### PR TITLE
[stable/presto] Be able to configure connectors

### LINKED ISSUE #19831: Presto for interactive SQL
Our requirement to use presto to access COS data. We have installed presto helm chart on our k8s cluster but not able to configure COS connection with it. Could you please help me to understand what are the steps need to be followed for using presto for interactive SQL connecting to COS data. 

```
kubectl get pod 

my-release-presto-coordinator-6f64697bbc-llb7p      2/2     Running     0          9d
my-release-presto-worker-6857797df8-9qw6b           2/2     Running     6          9d
my-release-presto-worker-6857797df8-bbhp4           2/2     Running     6          9d
my-release-presto-worker-6857797df8-nwv9p           2/2     Running     6          9d
my-release-presto-worker-6857797df8-qgf9x           2/2     Running     0          9d

kubectl exec -it my-release-presto-worker-6857797df8-9qw6b -- /bin/bash               

presto@my-release-presto-worker-6857797df8-9qw6b:/$ presto
presto> show catalog
     -> ;
Error running command: Error starting query at http://localhost:8080/v1/statement returned an invalid response: JsonResponse{statusCode=404, statusMessage=Not Found, headers={cache-control=[must-revalidate,no-cache,no-store], content-length=[244], content-type=[text/html;charset=iso-8859-1], date=[Thu, 02 Jan 2020 12:59:49 GMT]}, hasValue=false} [Error: <html>
<head>
<meta http-equiv="Content-Type" content="text/html;charset=utf-8"/>
<title>Error 404 Not Found</title>
</head>
<body><h2>HTTP ERROR 404</h2>
<p>Problem accessing /v1/statement. Reason:
<pre>    Not Found</pre></p>
</body>
</html>
]

```

### PR DESCRIPTION
**Pull request by shawnzhu**
#### Is this a new chart
No.

#### What this PR does / why we need it:
It can not configure presto connectors at all in current presto chart v0.1.1. Which makes this chart less useful when deploying presto (which depends connectors to function).
This PR enables configuration options to specify connectors as inline values or a k8s secret.

#### Which issue this PR fixes
fixes #19831

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

File: stable/presto/templates/deployment-coordinator.yaml [hunk 1]
```diff
@@ -25,6 +25,15 @@ spec:
         - name: config-volume
           configMap:
             name: {{ template "presto.coordinator" . }}
+        {{- if .Values.server.catalog.secretName }}
+        - name: catalog-volume
+          secret:
+            secretName: {{ .Values.server.catalog.secretName }}
+        {{- else if .Values.server.catalog.data | quote }}
```
Comment (emmadickson) [inline stable/presto/templates/deployment-coordinator.yaml]: Why do you need this else condition here. Under what circumstance would the secret location change or not be what is specified in the first conditional.

File: stable/presto/values.yaml [hunk 2]
```diff
@@ -19,6 +19,14 @@ server:
       type: "UseG1GC"
       g1:
         heapRegionSize: "32M"
+  catalog: {}
```
Comment (emmadickson) [inline stable/presto/values.yaml]: Couldn't this have something naming the secret catalog-volume?

File: stable/presto/templates/deployment-coordinator.yaml (hunk 1 above)
Comment (shawnzhu) [reply stable/presto/templates/deployment-coordinator.yaml]: it provides the options to configures connectors via either k8s secret or configmap.

File: stable/presto/values.yaml (hunk 2 above)
Comment (shawnzhu) [reply stable/presto/values.yaml]: this is about backward compatibility so that it won't bring break change once upgrade.

**Review by emmadickson — APPROVED** [review]
LGTM!

**Comment by zanhsieh** [comment]
@shawnzhu 
Please bump this chart to latest version. Thank you.

**Comment by shawnzhu** [comment]
> Please bump this chart to latest version. Thank you.

@zanhsieh see https://github.com/helm/charts/pull/20646/commits/1080359bd04899cdb594da43d88cfa0075d1bb5c

File: stable/presto/templates/configmap-catalog.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+metadata:
+  name: {{ template "presto.catalog" . }}
+  labels:
+    app: {{ template "presto.name" . }}
+    chart: {{ template "presto.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+data:
+  {{ .Values.server.catalog.data }}
+---
```
Comment (zanhsieh) [inline stable/presto/templates/configmap-catalog.yaml]: I0429 16:29:02.853] [ERROR] templates/configmap-catalog.yaml: unable to parse YAML
I0429 16:29:02.853] 	error converting YAML to JSON: yaml: line 13: did not find expected key

Comment (shawnzhu) [reply stable/presto/templates/configmap-catalog.yaml]: I've fixed it in below commits. Thanks for point it out

**Comment by edwardflo** [comment]
@shawnzhu Any idea when this will be merged?

**Comment by shawnzhu** [comment]
> Any idea when this will be merged?

@edwardflo see #23539 as a follow up

**Pull request closed without merging**